src/lib/comms/sol-coap.c

```diff
@@ -662,8 +662,11 @@ pending_timeout_cb(void *data)
     struct pending_reply *reply = data;
     struct sol_coap_server *server = reply->server;
 
-    if (reply->cb((void *)reply->data, server, NULL, NULL))
-        return true;
+    if (reply->cb((void *)reply->data, server, NULL, NULL)) {
+        if (reply->freshness != PENDING_REPLY_FREE_ME) {
+            return true;
+        }
+    }
 
     sol_ptr_vector_remove(&server->pending, reply);
```

Comment:
No, because if you call ```pending_reply_free()``` while ```reply->freshness``` is set to ```PENDING_REPLY_PENDING```, it won't free, it'll just set ```reply->freshness``` to ```PENDING_REPLY_FREE_ME```. Have a [look](https://github.com/solettaproject/soletta/pull/2055/files#diff-71a300ca520dd6eed36cfceb353e8312R643).